### PR TITLE
update katello gem deps

### PR DESCRIPTION
**Pull request by evgeni**

### CODE REVIEW

**Comment by snecklifter** [comment]
```Gem loading error: Could not find 'pulp_2to3_migration_client' (< 0.3.0, > 0.02b2) - did find: [pulp_2to3_migration_client-0.2.0b2]

**Comment by evgeni** [comment]
[test rpm]

File: packages/katello/rubygem-katello/rubygem-katello.spec [hunk 1]
```diff
@@ -22,9 +22,10 @@ URL:     https://theforeman.org/plugins/katello
 Source0: https://rubygems.org/downloads/%{gem_name}-%{version}%{?prerelease}.gem
 
 Requires: foreman-postgresql
+Requires: foreman < %{foreman_max_version}
+BuildRequires: %{?scl_prefix}rubygem(pg)
```
Comment (evgeni) [inline packages/katello/rubygem-katello/rubygem-katello.spec:26]: @jlsherrill seems this was introduced by https://github.com/Katello/katello/commit/514770541975012ea85a5780fd67c237cb535e2d but never added a dependency to the gemspec

Comment (jlsherrill) [reply packages/katello/rubygem-katello/rubygem-katello.spec:26]: ahh, did not realize this was a build time dep now, but it makes sense

Comment (evgeni) [reply packages/katello/rubygem-katello/rubygem-katello.spec:26]: https://github.com/Katello/katello/pull/8757

**Comment by evgeni** [comment]
repoclosure because I didn't mash :see_no_evil:

**Pull request merged**